### PR TITLE
Switch from SizeMe to ResizeDetector.

### PR DESCRIPTION
**Pull request by rpiaggio**
Also, resizer logic was moved to a `div` in `Routing`.

For the moment the facade is here but it will eventually be moved a standalone project.

(And cats update).

### CODE REVIEW

File: common/src/main/scala/explore/components/Tile.scala [hunk 1]
```diff
@@ -48,9 +48,9 @@ object Tile {
       .builder[Props]
       .stateless
       .render_PC { (p, c) =>
-        SizeMe(monitorHeight = true) { s =>
+        ResizeDetector() { s =>
```
Comment (rpiaggio) [inline common/src/main/scala/explore/components/Tile.scala:51]: ResizeDetector tracks height by default.

File: explore/src/main/scala/explore/tabs/ObsTabContents.scala [hunk 2]
```diff
@@ -339,7 +330,10 @@ object ObsTabContents {
         }
       )
       .renderBackend[Backend]
-      .componentDidMount($ => $.backend.readWidthPreference *> $.backend.readLayoutPreference)
+      .componentDidMount($ =>
+        $.backend.readWidthPreference($.props.userId.get) *>
```
Comment (cquiroz) [inline explore/src/main/scala/explore/tabs/ObsTabContents.scala:334]: This is unrelated. Would it be possible to combine these into one call?

**Review by cquiroz — APPROVED** [review]
Looks great, and great work hunting this bug
